interpretation/polls.py

```diff
@@ -0,0 +1,166 @@
+            self.emoji = emoji
+            self.votes = int(votes)
+
+class Disqualified:
+    def __init__(self,user_id,reason = 0):
+        self.user = user_id
+
+        self.reason = reason
+        # 0 - double voting
+        # 1 - self voting
+
+def count_votes(voting_table, purpose = 'lynch', mayor = 0):
+    """Count votes based on reactions given to """
+    user_table = []
+    blacklist = []
+    blacklist2 = []
+    emoji_table = []
+
+    for vote in voting_table:
+        # Add new emoji
+        if vote[1] not in emoji_table:
+            emoji_table.append(vote[1])
+
+        # Filter self-votes and double votes
+        if vote[0] in blacklist or vote[0] in blacklist2:
+            pass
+        elif vote[0] == emoji_to_player(vote[1]) and purpose not in ['Mayor','Reporter']:
+            blacklist2.append(vote[0])
+            if vote[0] in user_table:
+                user_table.remove(vote[0])
+        elif vote[0] in user_table:
+            blacklist.append(vote[0])
+            user_table.remove(vote[0])
+        else:
+            user_table.append(vote[0])
+
+    # Evaluate table, filter double votes and continue
+    voting_table = [Vote(vote[0],vote[1],int(db_get(vote[0],'votes'))) for vote in voting_table if vote[0] in user_table]
+    blacklist = [Disqualified(user) for user in blacklist]
+    blacklist2 = [Disqualified(user,1) for user in blacklist2]
+    blacklist.extend(blacklist2)
+
+    # Add extra (secret) outside lynches, may some special events happen.
+    if purpose == 'lynch':
+        for user in player_list():
+            if int(db_get(user,'threatened')) > 0:
+                print('{} has been threatened!'.format(user))
+                voting_table.append(Vote("RAVEN THREAT",db_get(user,'emoji'),int(db_get(user,'threatened'))))
+
+                # Add emoji to table
+                if db_get(user,'emoji') not in emoji_table:
+                    emoji_table.append(db_get(user,'emoji'))
+
+    # print([[vote.user,vote.emoji,vote.votes] for vote in voting_table])
+
+    # Create the evaluation messages
+    answer = "**__Poll results "
+    log = "**__Poll results "
+
+    if purpose == 'wolf':
+        answer += "from wolf attack:__**\n\n"
+        log += "from wolf attack:__**\n\n"
+    elif purpose == 'cult':
+        answer += "from cult attack:__**\n\n"
+        log += "from cult attack:__**\n\n"
+    elif purpose == 'thing':
+        answer += "from the swamp:__**\n\n"
+        log += "from the swamp:__**\n\n"
+    elif purpose == 'lynch':
+        answer += "from public execution on day {}:__**\n\n".format(day_number())
+        log += "from public execution on day {}:__**\n\n".format(day_number())
+    elif purpose == 'Mayor':
+        answer += "from Mayor election:__**\n\n"
+        log += "from Mayor election:__**\n\n"
+    elif purpose == 'Reporter':
+        answer += "from Reporter election:__**\n\n"
+        log += "from Reporter election:__**\n\n"
+
```

Comment:
I don't particularly like the fact that the above lines are hardcoded in, but I suppose it gets the job done

Reply:
I don't feel comfortable with it either, but I couldn't think of any method that would not become overly complicated or oddly specific.